test/syscalls/linux/pty.cc

```diff
@@ -650,6 +650,25 @@ class PtyTest : public ::testing::Test {
     t.c_lflag |= ICANON;
     EXPECT_THAT(ioctl(replica_.get(), TCSETS, &t), SyscallSucceeds());
   }
+ 
+  // Writes master_input to the master file descriptor and verifies that
+  // the replica and the echo output match what is expected.
+  void TestCanonicalIO(const char* master_input, const char* expected_replica_output, const char* expected_echo_output) {
```

Comment:
nit: The indentation here is wrong (should have no indentation at all). I believe this may trip some of the internal presubmit checks. Can you run clang-format?

Reply:
silly question, but how should I run `clang-format`? Do I just run `clang-format -i pty.cc`? It made some changes to other parts of this file as well; do you want me to commit those too?

Reply:
Yes, that should be fine.

Reply:
Clang is also complaining that `no header providing "strlen" is directly included`. https://clang.llvm.org/extra/clang-tidy/checks/misc/include-cleaner.html I think you need `string.h`.

Reply:
I pushed the changes. Thanks!

Reply:
Now it complains about the same lack of header for `std::string` (:
I think you need `#include <string>`.

Reply:
Oops, sorry, fixed this as well. How do you run `clang-tidy` with the proper dependencies included / is there a recommended dev setup for gvisor?